optaplanner-docs/src/main/asciidoc/Chapter-Optimization_algorithms/Chapter-Optimization_algorithms.adoc

```diff
@@ -541,15 +536,15 @@ Alternatively you can use AND, for example: terminate after reaching a feasible
   </termination>
 ----
 
-This example ensures it doesn't just terminate after finding a feasible solution, but also completes any obvious improvements on that solution before terminating.
+This example ensures it does not just terminate after finding a feasible solution, but also completes any obvious improvements on that solution before terminating.
 
 
 [[asynchronousTermination]]
 === Asynchronous Termination from Another Thread
 
-Sometimes you'll want to terminate a Solver early from another thread, for example because a user action or a server restart.
-This cannot be configured by a `Termination` as it's impossible to predict when and if it will occur.
-Therefore the `Solver` interface has these 2 thread-safe methods:
+Asychronous termination from another thread occurs When a `Solver` needs to be terminated early from another thread, for example, due to a user action or a server restart.
```

Comment:
As above. I can't change it myself in the online git editor but can create a new pull request if necessary.